src/coreclr/jit/lsra.h

```diff
@@ -486,6 +487,12 @@ class RegRecord : public Referenceable
         {
             registerType = FloatRegisterType;
         }
+#if defined(TARGET_XARCH) && defined(FEATURE_SIMD)
+        else if (emitter::isMaskReg(reg))
```

Comment:
Given that `isGeneralRegister` is going to be the most common case, I think we probably want to factor this code a bit differently so that the most common path has the least checks.

Something like:
```
if (emitter::isGeneralRegister(reg))
{
    // The constructor defaults to IntRegisterType
    assert(registerType == IntRegisterType);
}
else if (emitter::isFloatReg(reg))
{
    registerType = FloatRegisterType;
}
else
{
    assert(emitter::isMaskReg(reg));
    registerType = MaskRegisterType;
}
```